src/lib/y2network/interface_config_builders/ctc.rb

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface_config_builder"
+
+Yast.import "LanItems"
+Yast.import "NetworkInterfaces"
+
+module Y2Network
+  module InterfaceConfigBuilders
+    class Ctc < InterfaceConfigBuilder
+      extend Forwardable
+
+      def initialize(config: nil)
+        super(type: InterfaceType::CTC, config: config)
+      end
+
+      def_delegators :@connection_config,
+        :read_channel, :read_channel=,
+        :write_channel, :write_channel=,
+        :protocol, :protocol=
+
+      def device_id
+        return if read_channel.to_s.empty?
+
+        "#{read_channel}:#{write_channel}"
+      end
+
+      def device_id_from(channel)
+        cmd = "/sbin/lszdev ctc -c id -n".split(" ")
+
+        Yast::Execute.stdout.on_target!(cmd).split("\n").find do |d|
+          d.include? channel
+        end
+      end
+
+      def configure
+        cmd = "/sbin/chzdev ctc #{device_id} -e protocol=#{protocol}".split(" ")
+
+        Yast::Execute.on_target!(*cmd, allowed_exitstatus: 0..255).zero?
+      end
+
+      def configured_interface
+        cmd = "/sbin/lszdev #{device_id} -c names -n".split(" ")
+
+        Yast::Execute.stdout.on_target!(cmd).chomp
+      end
```

Comment:
I am not sure if it should be in builder as builder job is mainly be gateway between backend and frontend. So it would be nicer for me to have dedicated class for this commands and its parsing.

Reply:
ok, so S390DeviceActivator as the base class and an activator per type is fine with you?

Reply:
@teclator yes, sounds good for me.